apps/src/dance/ai/AiGenerateWeights.py

```diff
@@ -0,0 +1,150 @@
+import spacy
+import json
+
+# Load the spaCy model
+nlp = spacy.load("en_core_web_md")
+
+# Define your "id" and "palettes" lists
+id = ["poopy", "romantic", "party", "silly", "sparkle", "happy", "magic", "spooky", "cute", "funky", "wavy", "lights", "rainbow", "robot", "chaotic", "disco", "zen", "fast", "evil", "cold", "cosmic", "sad", "black-and-white", "warm", "cool"]
+palettes = ["cool", "electronic", "ice cream", "neon", 'rave', "tropical","vintage",'warm']
+backgrounds = ['circles',
+    'color',
+    'diamonds',
+    'disco',
+    'fireworks',
+    'swirl',
+    'kaleidoscope',
+    'lasers',
+    'splatter',
+    'rainbow',
+    'snowflakes',
+    'text',
+    'galaxy',
+    'sparkles',
+    'spiral',
+    'disco',
+    'stars',
+    'music',
+    'ripples',
+    'random',
+    'quads',
+    'flowers',
+    'squiggles',
+    'stars',
+    'petals',
+    'clouds',
+    'grid',
+  ]
+
+foregrounds = [
+    'bubbles',
+    'confetti',
+    'hearts',
+    'music',
+    'pineapples',
+    'pizzas',
+    'poop',
+    'rain',
+    'rainbows',
+    'smile',
+    'spotlight',
+    'lights',
+    'tacos',
+    'emojis',
+    'hearts',
+    'stars',
+    'paint',
+  ]
+
+palettedict = {}
+bgdict = {}
+fgdict = {}
+# Calculate and print similarity scores
+for id_word in id:
+    palette_scores = []
+    for palette_word in palettes:
+        id_token = nlp(id_word)
+        palette_token = nlp(palette_word)
+        similarity_score = id_token.similarity(palette_token)
+        palette_scores.append(round(similarity_score, 2))
+    
+    bg_scores = []
+    for bg_word in backgrounds:
+        id_token = nlp(id_word)
+        bg_token = nlp(bg_word)
+        similarity_score = id_token.similarity(bg_token)
+        bg_scores.append(round(similarity_score, 2))
+        
+    fg_scores = []
+    for fg_word in foregrounds:
+        id_token = nlp(id_word)
+        fg_token = nlp(fg_word)
+        similarity_score = id_token.similarity(fg_token)
+        fg_scores.append(round(similarity_score, 2))
+        
+    palettedict[id_word] = palette_scores
+    bgdict[id_word] = bg_scores
+    fgdict[id_word] = fg_scores
+
+paletteoutput = {'emojiAssociations': palettedict, 'output': palettes}
+bgoutput = {'emojiAssociations': bgdict, 'output': backgrounds}
+fgoutput = {'emojiAssociations': fgdict, 'output': foregrounds}
+
+# Modify output to reflect true file names
+bgoutput['output'] = [
```

Comment:
For my understanding, what's the difference between this list and `backgrounds`?

Reply:
This was built to help address situations where "growing_stars" would return with vectors consisting entirely of [0s] as spaCy would interpret underscored phrases such as "growing_stars" as a unique word that didn't exist in any context (based off scraped internet data).

I compromised this by selecting what I perceived as the most relevant word from each phrase (e.g. disco instead of disco ball, stars instead of growing stars) and using that as the token that would be compared against the emoji for similarity.

I originally also contemplated trying to do something more programmatic (e.g. string split on _, take [0]) but there wasn't any uniformity between which side of the underscore had more context-rich words than others (e.g. "smiling_poop" should have "poop" extracted vs "hearts_colorful" with "hearts" being more impactful)

These were then converted back into their actual output-ids as stored within our dataset to make sure these outputs were readable by our typescript.

Reply:
Gotcha, thanks for the explanation! It would probably be good to have some single source of truth for these in that case to associate the spaCy-friendly names with the actual effect name (e.g. in a JSON file somewhere) so that we don't risk messing up the order in case we add/remove a single effect. cc: @fisher-alice in case this is similar to the work you're doing to read the effect names from the blocks?

Reply:
Thanks for the heads up! This is a pending [ticket](https://trello.com/c/QpakSqaK/55-add-helper-function-to-return-lists-of-background-foreground-effects-and-palettes).